src/System.Runtime.Extensions/src/System/Globalization/Extensions.cs

```diff
@@ -26,71 +28,12 @@ public static StringComparer GetStringComparer(this CompareInfo compareInfo, Com
                 return StringComparer.OrdinalIgnoreCase;
             }
 
-            if ((options & CultureAwareComparer.ValidCompareMaskOffFlags) != 0)
+            if ((options & ValidCompareMaskOffFlags) != 0)
             {
                 throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
             }
 
-            return new CultureAwareComparer(compareInfo, options);
-        }
-    }
-
-    internal sealed class CultureAwareComparer : StringComparer
-    {
-        internal const CompareOptions ValidCompareMaskOffFlags =
-            ~(CompareOptions.IgnoreCase | CompareOptions.IgnoreSymbols | CompareOptions.IgnoreNonSpace |
-              CompareOptions.IgnoreWidth | CompareOptions.IgnoreKanaType | CompareOptions.StringSort);
-
-        private readonly CompareInfo _compareInfo;
-        private readonly CompareOptions _options;
-
-        internal CultureAwareComparer(CompareInfo compareInfo, CompareOptions options)
-        {
-            Debug.Assert((options & ValidCompareMaskOffFlags) == 0);
-            _compareInfo = compareInfo;
-            _options = options;
-        }
-
-        public override int Compare(string x, string y)
-        {
-            if (Object.ReferenceEquals(x, y)) return 0;
-            if (x == null) return -1;
-            if (y == null) return 1;
-            return _compareInfo.Compare(x, y, _options);
-        }
-
-        public override bool Equals(string x, string y)
-        {
-            if (Object.ReferenceEquals(x, y)) return true;
-            if (x == null || y == null) return false;
-
-            return (_compareInfo.Compare(x, y, _options) == 0);
-        }
-
-        public override int GetHashCode(string obj)
-        {
-            if (obj == null)
-            {
-                throw new ArgumentNullException(nameof(obj));
-            }
-
-            // StringSort used in compare operation and not with the hashing
-            return _compareInfo.GetHashCode(obj, _options & (~CompareOptions.StringSort));
-        }
-
-        // Equals method for the comparer itself. 
-        public override bool Equals(object obj)
-        {
-            CultureAwareComparer comparer = obj as CultureAwareComparer;
-            return
-                comparer != null &&
-                _options == comparer._options &&
-                _compareInfo.Equals(comparer._compareInfo);
-        }
-
-        public override int GetHashCode()
-        {
-            return _compareInfo.GetHashCode() ^ ((int)_options & 0x7FFFFFFF);
+            return StringComparer.Create(CultureInfo.GetCultureInfo(compareInfo.Name), options);
```

Comment:
It still does not feel right to me to make this method many times slower with this change.

Reply:
@tarekgh any other things that we can do ?

Reply:
> It still does not feel right to me to make this method many times slower with this change.

I cannot think of any ready better solution rather than returning back the CultureAwareComparer here. the best we can do is to expose StringCompare.Create which takes a compare info.

Reply:
> move the whole extension class to coelib

That would be my pick.

Reply:
@Anipik could you please move this extension class to corelib? thanks a lot for your effort here. 

@jkotas do you want to keep this PR open till we move the extension class to corelib? or we can proceed with this PR and have another PR to fix that?

Another thing for this PR, could you please clean up the ref folder too corefx\src\System.Globalization.Extensions\ref?

Reply:
@jkotas should we move to corelib in this PR or in a different PR ?

Reply:
@Anipik You may start creating the coreclr PR from now.